lib/presentation/mixins/connect_page_mixin.dart

```diff
@@ -286,4 +301,137 @@ mixin ConnectPageMixin {
+        return clientExisted;
+      } else {
+        return null;
+      }
+    } catch (e) {
+      Logs().e(
+        'ConnectPageMixin::getClientExisted: Exception: $e',
+      );
+      return null;
+    }
+  }
+
+  Future<void> logoutAction({required MatrixState matrix}) async {
+    Logs().d('ConnectPageMixin::logoutAction');
+    await _tryToUploadKeyBackup(matrix: matrix);
+
+    if (PlatformInfos.isMobile) {
+      await _logoutActionsOnMobile(matrix: matrix);
+    } else {
+      await _logoutActionOnOtherPlatform(matrix: matrix);
+    }
+  }
+
+  Future<void> _tryToUploadKeyBackup({required MatrixState matrix}) async {
+    Logs().d('ConnectPageMixin::_tryToUploadKeyBackup');
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        await matrix.client.encryption?.keyManager.uploadInboundGroupSessions();
+      },
+    );
+  }
+
+  Future<void> _logoutActionsOnMobile({required MatrixState matrix}) async {
+    await tryLogoutSso(matrix: matrix);
+
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        try {
+          await _clearAndSignOut(matrix: matrix);
+        } catch (e) {
+          Logs().e('ConnectPageMixin()::_logoutActionsOnMobile - error: $e');
+        }
+      },
+    );
+  }
+
+  Future<void> _deleteTomConfigurations({required MatrixState matrix}) async {
+    try {
+      final userId = matrix.activatedUserId;
+      Logs().d(
+        'ConnectPageMixin::_deleteTomConfigurations - Client ID: $userId',
+      );
+      if (matrix.twakeSupported) {
+        await getIt
+            .get<ToMConfigurationsRepository>()
+            .deleteTomConfigurations(userId);
+      }
+      Logs().d(
+        'ConnectPageMixin::_deleteTomConfigurations - Success',
+      );
+    } catch (e) {
+      Logs().e(
+        'ConnectPageMixin::_deleteTomConfigurations - error: $e',
+      );
+    }
+  }
+
+  Future<void> _deleteFederationConfigurations({
+    required MatrixState matrix,
+  }) async {
+    try {
+      final userId = matrix.activatedUserId;
+      Logs().d(
+        'ConnectPageMixin::_deleteFederationConfigurations - Client ID: $userId',
+      );
+      await getIt
+          .get<FederationConfigurationsRepository>()
+          .deleteFederationConfigurations(userId);
+      Logs().d(
+        'ConnectPageMixin::_deleteFederationConfigurations - Success',
+      );
+    } catch (e) {
+      Logs().e(
+        'ConnectPageMixin::_deleteFederationConfigurations - error: $e',
+      );
+    }
+  }
+
+  Future<void> _logoutActionOnOtherPlatform({
+    required MatrixState matrix,
+  }) async {
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        try {
+          await _clearAndSignOut(matrix: matrix);
+        } catch (e) {
+          Logs().e(
+            'ConnectPageMixin()::_logoutActionOnOtherPlatform - error: $e',
+          );
+        } finally {
+          await tryLogoutSso(matrix: matrix);
```

Comment:
Should be inside `try-catch` block. A lot of kinds of error with this SSO logout.

Reply:
Inside `tryLogoutSso` already used `try-catch` block